compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py

```diff
@@ -0,0 +1,330 @@
+
+
+# In[2]:
+
+
+total_candidates_df = (
+    pd
+    .read_table("../dataset_statistics/results/all_ctd_candidates.tsv.xz")
+    .sort_values("candidate_id")
+)
+total_candidates_df.head(2)
+
+
+# In[3]:
+
+
+sentence_prediction_df = (
+    pd
+    .read_table("results/all_predicted_sentences.tsv")
+    .sort_values("candidate_id")
+)
+sentence_prediction_df.head(2)
+
+
+# In[4]:
+
+
+total_candidates_pred_df = (
+    total_candidates_df[[
+    "doid_id", "doid_name", 
+    "drugbank_id", "drug_name", 
+    "text", "hetionet",
+    "candidate_id"
+    ]]
+    .merge(sentence_prediction_df, on="candidate_id")
+)
+total_candidates_pred_df.head(2)
+
+
+# In[5]:
+
+
+grouped_candidates_pred_df=(
+    total_candidates_pred_df
+    .groupby(["doid_id", "drugbank_id"], as_index=False)
+    .agg({
+        "model_prediction": ['max', 'mean', 'median'], 
+        'hetionet': 'max',
+        "drug_name": 'first',
+        "doid_name": 'first'
+    })
+)
+grouped_candidates_pred_df.head(2)
+
+
+# In[6]:
+
+
+grouped_candidates_pred_df.columns = [
+    "_".join(col) 
+    if col[1] != '' and col[0] not in ['hetionet', 'drug_name', 'doid_name'] else col[0] 
+    for col in grouped_candidates_pred_df.columns.values
+]
+grouped_candidates_pred_df.head(2)
+
+
+# In[7]:
+
+
+grouped_candidates_pred_df.to_csv(
+    "results/combined_predicted_ctd_sentences.tsv.xz", 
+    index=False, sep="\t", compression="xz"
+)
+
+
+# In[8]:
+
+
+grouped_candidates_pred_df.hetionet.value_counts()
+
+
+# In[9]:
+
+
+performance_map = {}
+
+
+# In[10]:
+
+
+precision, recall, pr_threshold = precision_recall_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_max,
+)
+
+fpr, tpr, roc_threshold = roc_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_max,
+)
+
+performance_map['max'] = {
+    "precision":precision, "recall":recall, 
+    "pr_threshold":pr_threshold, "false_pos":fpr,
+    "true_pos":tpr, "roc_threshold":roc_threshold, 
+}
+
+
+# In[11]:
+
+
+precision, recall, pr_threshold = precision_recall_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_mean,
+)
+
+fpr, tpr, roc_threshold = roc_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_mean,
+)
+
+performance_map['mean'] = {
+    "precision":precision, "recall":recall, 
+    "pr_threshold":pr_threshold, "false_pos":fpr,
+    "true_pos":tpr, "roc_threshold":roc_threshold, 
+}
+
+
+# In[12]:
+
+
+precision, recall, pr_threshold = precision_recall_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_median,
+)
+
+fpr, tpr, roc_threshold = roc_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_median,
+)
+
+performance_map['median'] = {
+    "precision":precision, "recall":recall, 
+    "pr_threshold":pr_threshold, "false_pos":fpr,
```

Comment:
What is the definition of this pr_threshold?  How is it determined?

Reply:
pr_threshold -> precision recall threshold. This is determined through sklearns `precision_recall_curve` function.